lib/puppet/provider/gluster_volume/gluster_volume.rb

```diff
@@ -25,6 +25,17 @@ def self.prefetch(resources)
     end
   end
 
+  def missing_peers
+    # First get the hostnames of all the remote peers we know about.
+    peers = self.class.all_peers
+    # Then add our own addresses so we know the local machine isn't missing.
+    [:hostname, :fqdn, :ipaddress].each { |f| peers << Facter.value(f) }
```

Comment:
We could add a `local_peer_addresses` parameter or something, I suppose. We should probably also rename `ignore_peers` in the peer resource. That can happen in another branch, though.